spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,649 @@
+        void shouldExportRegeneratedEntitlementCertificates() throws Exception {
+            int target = ENTITILEMENT_CERTIFICATES_PATH.length() - 1;
+            List<ZipEntry> entitlementCerts = export.stream()
+                .filter(entry -> entry.getName().startsWith(ENTITILEMENT_CERTIFICATES_PATH))
+                .filter(entry -> entry.getName().lastIndexOf('/') == target)
+                .collect(Collectors.toList());
+
+            // Regenerate some entitlement certificates, and generate a new manifest:
+            String consumerUuid = consumer.getUuid();
+            consumerApi.regenerateEntitlementCertificates(consumerUuid, null, true);
+            File newManifest = createExport(client, consumerUuid, cdn);
+            ZipFile newExport = ExportUtil.getExportArchive(newManifest);
+            AtomicInteger newEntitlementsSize = new AtomicInteger(0);
+            newExport.stream()
+                .filter(entry -> entry.getName().startsWith(ENTITILEMENT_CERTIFICATES_PATH))
+                .filter(entry -> entry.getName().lastIndexOf('/') == target)
+                .forEach(entry -> {
+                    // Cert filenames should be completely different now
+                    assertFalse(entitlementCerts.contains(entry));
+                    newEntitlementsSize.incrementAndGet();
+                });
+
+            assertEquals(entitlementCerts.size(), newEntitlementsSize.get());
+        }
+
+        @Test
+        @DisplayName("should export rules")
+        void shouldExportRules() throws Exception {
+            ZipEntry rulesFile = export.getEntry(RULES_PATH + "rules.js");
+            assertNotNull(rulesFile);
+            String expectedRules = rulesApi.getRules();
+            Base64 base64 = new Base64();
+            expectedRules = new String(base64.decode(expectedRules.getBytes()));
+
+            try (InputStream istream = export.getInputStream(rulesFile)) {
+                String actualRules = new String(istream.readAllBytes());
+                assertEquals(expectedRules, actualRules);
+            }
+        }
+
+        @Test
+        @DisplayName("should export products")
+        void shouldExportProducts() throws Exception {
+            int target = PRODUCTS_PATH.length() - 1;
+            AtomicInteger productCount = new AtomicInteger(0);
+            export.stream()
+                .filter(entry -> entry.getName().startsWith(PRODUCTS_PATH))
+                .filter(entry -> entry.getName().lastIndexOf('/') == target)
+                .filter(entry -> entry.getName().endsWith(".json"))
+                .forEach(entry -> {
+                    try {
+                        productCount.incrementAndGet();
+                        String name = new File(entry.getName()).getName().replace(".json", "");
+                        ProductDTO expected = productIdToProduct.get(name);
+                        assertNotNull(expected);
+                        ProductDTO actual = ExportUtil
```

Comment:
No check on the output of this call. Should add an assertNotNull following it.

Reply:
Added